src/main/java/org/jabref/logic/importer/fetcher/Medra.java

```diff
@@ -0,0 +1,175 @@
+package org.jabref.logic.importer.fetcher;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PushbackInputStream;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.jabref.logic.importer.FetcherException;
+import org.jabref.logic.importer.IdBasedParserFetcher;
+import org.jabref.logic.importer.ParseException;
+import org.jabref.logic.importer.Parser;
+import org.jabref.logic.importer.util.JsonReader;
+import org.jabref.logic.importer.util.MediaTypes;
+import org.jabref.logic.net.URLDownload;
+import org.jabref.model.entry.AuthorList;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.StandardField;
+import org.jabref.model.entry.types.EntryType;
+import org.jabref.model.entry.types.StandardEntryType;
+
+import kong.unirest.json.JSONArray;
+import kong.unirest.json.JSONException;
+import kong.unirest.json.JSONObject;
+import org.apache.http.client.utils.URIBuilder;
+
+/**
+ * A class for fetching DOIs from Medra
+ *
+ * @see <a href="https://data.medra.org">mEDRA Content Negotiation API</a> for an overview of the API
+ * <p>
+ * It requires "Accept" request Header attribute to be set to desired content-type.
+ */
+public class Medra implements IdBasedParserFetcher {
+
+    public static final String API_URL = "https://data.medra.org";
+
+    @Override
+    public String getName() {
+        return "mEDRA";
+    }
+
+    @Override
+    public Parser getParser() {
+        return inputStream -> {
+            JSONObject response = JsonReader.toJsonObject(inputStream);
+
+            List<BibEntry> entries = new ArrayList<>();
+            BibEntry entry = jsonItemToBibEntry(response);
+            entries.add(entry);
+
+            return entries;
+        };
+    }
+
+    private BibEntry jsonItemToBibEntry(JSONObject item) throws ParseException {
+        try {
+            BibEntry entry = new BibEntry();
+            entry.setType(convertType(item.getString("type")));
```

Comment:
Could you do a chaining of `setField`?

```java
  return new BibEntry(convertType(item.getString("type")))
    .withField(StandardField.TITLE, item.getString("title"))
    .withField...
    ...
```

Reply:
Done.

Reply:
That comment also applied for the test cases. I did that in https://github.com/mind000/jabref/pull/1/commits/9b9b14f2871dceddf4e9a8f7c748b668c5c97f15 to show what I meant.